src/tools/snapshot/nodes/ArraySerializationNode.java

```diff
@@ -0,0 +1,144 @@
+package tools.snapshot.nodes;
+
+import java.nio.ByteBuffer;
+
+import som.interpreter.Types;
+import som.vm.constants.Classes;
+import som.vmobjects.SArray;
+import tools.snapshot.SnapshotBuffer;
+
+
+public class ArraySerializationNode extends AbstractSerializationNode {
+  private static final byte TYPE_BOOLEAN = 0;
+  private static final byte TYPE_DOUBLE  = 1;
+  private static final byte TYPE_LONG    = 2;
+  private static final byte TYPE_OBJECT  = 3;
+  private static final byte TYPE_EMPTY   = 4;
+
+  @Override
+  public void serialize(final Object o, final SnapshotBuffer sb) {
+    assert o instanceof SArray;
+    SArray sa = (SArray) o;
+    serializeStorage(sa, sb);
+  }
+
+  protected void serializeStorage(final SArray sa, final SnapshotBuffer sb) {
+    int requiredSpace;
+    if (sa.isBooleanType()) {
```

Comment:
Since this looks like a node, shouldn't it simply specialize on the array type with a `@Specialization` and guard?